### ISSUE TITLE
v2 beta: Cancelling orders in Mirakl sets amount to 0

### ISSUE THREAD

**Issue by christophersjchow** [stripe/stripe-mirakl-connector#51]
When a logistic order is cancelled, the amount is set to 0 in Mirakl. This results in the connector incorrectly subtracting the amount from the total.

```
[2021-01-22 02:29:49] app.INFO: Deducting refused/canceled order amount 0 from final amount. {"commercial_id":"0593775609","order_id":"0593775609-A"} []
[2021-01-22 02:29:49] app.INFO: Deducting refused/canceled order amount 0 from final amount. {"commercial_id":"0593775609","order_id":"0593775609-B"} []
[2021-01-22 02:29:49] app.INFO: Deducting refused/canceled order amount 0 from final amount. {"commercial_id":"0593775609","order_id":"0593775609-C"} []
```

In the extreme case where all orders are cancelled the final logic to check if `$finalAmount > 0` also is incorrect and the connector ends up capturing the payment incorrectly.

**Comment by christophersjchow** [comment]
I'm also checking with Mirakl if this is expected behaviour.

**Comment by youssef-stripe** [comment]
Thanks for flagging @christophersjchow. That's because Mirakl deducts from "total_price" the line orders cancelled (unlike refused order lines) and adds the cancelled amounts under order_lines > cancelations. The connector should get the amount to deduct from this hash instead of total_price.

We will push a fix for this as well as soon as possible.

**Comment by youssef-stripe** [comment]
This should be fixed by 7a1f99f. Let me know if you still find discrepancies in the cancel flow.

**Comment by christophersjchow** [comment]
This still seems to be an issue in our testing after applying latest changes. The full authorisation is still captured in Stripe.
